upload/common/uploaded_file.py

```diff
@@ -22,11 +22,27 @@ class UploadedFile:
 
     @classmethod
     def create(cls, upload_area, checksums={}, name=None, content_type=None, data=None):
+        """
+        Check if the file exists already and if so, return it. Also return a boolean designating whether the file was
+        freshly uploaded or if the file was simply fetched. """
+        file_newly_created = True
+
+        paginator = s3.meta.client.get_paginator('list_objects')
```

Comment:
I don't think we should be checking if the file was created via s3. store_file is a synchronous call that needs to complete within the 30 second api gateway timeout. We should instead run a query against the database searching for this filename. If the filename exists in the db, we should then just retrieve a single object from s3 and compare the checksums.

Additionally,  I think the checksums passed in here is just the single client side checksum correct? We should just compare this client side checksum against the single checksum in the s3 object metadata rather than all the checksums.

Reply:
Alternatively you could just pass in the filename to the prefix so that only 1 object is returned at max? These buckets can get into the hundreds of thousands of files during exporting and thus want to avoid iterating through all the files

Reply:
I understand that different filenames can have different checksums (thus probably the initial approach at iterating through all the files). If we want to account for this case (which is arguable), we'd want to run the query against the checksums stored in the database

Reply:
Did your second suggestion which actually matches the logic I have in dcp-cli

Reply:
Hmm the code still looks like the old code? Is this something that still needs to be pushed up?

Reply:
Oops; pushed the changes now!